src/wallet/wallet.cpp

```diff
@@ -98,10 +98,7 @@ static std::set<CWallet*> g_unloading_wallet_set;
 // Custom deleter for shared_ptr<CWallet>.
 static void ReleaseWallet(CWallet* wallet)
 {
-    // Unregister and delete the wallet right after BlockUntilSyncedToCurrentChain
```

Comment:
I don't understand why this has to be removed. Can you add some more detail to the commit log?

(I think the current commit log "Scheduler thread servicing chain state is stopped before wallet unloading and can't be useful" isn't right. `ReleaseWallet()` is the delete expression for the CWallet shared_ptr, and so is called for example when the wallet is unloaded during runtime.)

Reply:
I've run tests again, can't remember but I had a test staling some point, I think I may have introduce a height error at some point which would false `waitForNotificationsIfTipChanged` and later corrected it without realizing it would fix this. Drop this commit.